languages/cpp/src/shared/src/Transport/MockTransport.h

```diff
@@ -0,0 +1,1087 @@
+        Firebolt::Error Invoke(const string &method, const PARAMETERS &parameters, RESPONSE &response)
+        {
+            std::cout << "Inside Mock Transport Invoke function" << std::endl;
+            Entry slot;
+            uint32_t id = _channel->Sequence();
+            Firebolt::Error result = Send(method, parameters, id);
+
+            WPEFramework::Core::JSONRPC::Message message;
+            message.Designator = method;
+            std::unique_ptr<JsonEngine> jsonEngine = std::make_unique<JsonEngine>();
+            result = jsonEngine->MockResponse(message, response);
+            FromMessage((INTERFACE *)&response, message);
+            return (result);
+        }
+#endif
+
+        template <typename PARAMETERS>
+        Firebolt::Error InvokeAsync(const string &method, const PARAMETERS &parameters, uint32_t &id)
+        {
+            Entry slot;
+            id = _channel->Sequence();
+            return Send(method, parameters, id);
+        }
+
+        template <typename RESPONSE>
+        Firebolt::Error WaitForResponse(const uint32_t& id, RESPONSE& response, const uint32_t waitTime)
+        {
+            int32_t result = WPEFramework::Core::ERROR_TIMEDOUT;
+            _adminLock.Lock();
+            typename PendingMap::iterator index = _pendingQueue.find(id);
+            Entry& slot(index->second);
+            _adminLock.Unlock();
+
+            if (slot.WaitForResponse(waitTime) == true) {
+                WPEFramework::Core::ProxyType<WPEFramework::Core::JSONRPC::Message> jsonResponse = slot.Response();
+
+                // See if we have a jsonResponse, maybe it was just the connection
+                // that closed?
+                if (jsonResponse.IsValid() == true) {
+                    if (jsonResponse->Error.IsSet() == true) {
+                        result = jsonResponse->Error.Code.Value();
+                    }
+                    else {
+                        result = WPEFramework::Core::ERROR_NONE;
+                        if ((jsonResponse->Result.IsSet() == true)
+                            && (jsonResponse->Result.Value().empty() == false)) {
+                            FromMessage((INTERFACE*)&response, *jsonResponse);
+                        }
+                    }
+                }
+            } else {
+                result = WPEFramework::Core::ERROR_TIMEDOUT;
+            }
+            _adminLock.Lock();
+            _pendingQueue.erase(id);
+            _adminLock.Unlock();
+            return FireboltErrorValue(result);
+        }
+
+        void Abort(uint32_t id)
+        {
+            _adminLock.Lock();
+            typename PendingMap::iterator index = _pendingQueue.find(id);
+            Entry &slot(index->second);
+            _adminLock.Unlock();
+            slot.Abort(id);
+        }
+
+        template <typename RESPONSE>
+        Firebolt::Error Subscribe(const string &eventName, const string &parameters, RESPONSE &response)
```

Comment:
Is this PR is based on the latest commit from this repo?

Reply:
Since this PR was outdated, moved changes to [PR 222](https://github.com/rdkcentral/firebolt-openrpc/pull/222) which is based off of latest `next`.